assets/js/practice.mjs

```diff
@@ -43,18 +43,67 @@ let selectedTime = 0;
 let char_you_typed = 0
 let characters;
 
+const getwpm = () => {
+    // access history from local storage
+    history = localStorage.getItem("typerHistory");
+    if (history != null) {
+        // parse history into array
+        historyArray = JSON.parse(history);
+        // initialize variables
+        let highestSpeed = 0;
+        let speed = 0;
+        // loop through history array to find highest speed
+        historyArray.forEach((item) => {
+            speed =  Math.ceil((item.char / 5) / (item.timeSession));
+            // if speed is higher than highest speed, set it as highest speed
+            if (speed > highestSpeed) {
+                highestSpeed = speed;
+            }
+        });
+        return highestSpeed;
+    } else {
+        // if user has no history, return 0
+        return 0;
+        
+    }
+};
+
+
 // making the quote from the words
 const makequote = () => {
-    // setting up difficulty level and quote length
-    if (easyLvlBtn.checked) {
-        selectedDifficultyLevel = "easy";
-        quoteLength = 50
-    } else if (interLvlBtn.checked) {
-        selectedDifficultyLevel = "medium";
-        quoteLength = 60
-    } else if (hardLvlBtn.checked) {
-        selectedDifficultyLevel = "hard";
-        quoteLength = 70
+
+    // get highest wpm of user
+    let wpm = getwpm();
+
+    // get time choosed by user
+    const time = getTime();
+
+    // if user has no history ie. new user
+    if (wpm == 0) {
+        // setting up default difficulty level for new user
+        if (easyLvlBtn.checked) {
+            selectedDifficultyLevel = "easy";
+            quoteLength = 70 * time
+        } else if (interLvlBtn.checked) {
+            selectedDifficultyLevel = "medium";
+            quoteLength = 60 * time
+        } else if (hardLvlBtn.checked) {
+            selectedDifficultyLevel = "hard";
+            quoteLength = 50 * time
```

Comment:
For Hard Level decrease the quote length as it is so large that the screen needs to be scrolled to see the text.
```suggestion
            quoteLength = 50 * time / 2
```

Reply:
Dividing it by 2 is not the solution here as it there will not be sufficient words to type in the timed session. So, I have reduced the default words to 40.